src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -410,3 +418,83 @@ export const duplicateForm = (
     },
   )
 }
+
+/**
+ * Updates form fields of given form depending on the given editFormFieldParams
+ * @param originalForm the original form to update form fields for
+ * @param editFormFieldParams the parameters stating the type of updates and the position metadata if update type is edit
+ *
+ * @returns ok(updated form) if form fields update successfully
+ * @returns err(EditFieldError) if invalid updates are being performed to form fields
+ * @returns err(set of DatabaseError) if any database errors occurs
+ */
+export const editFormFields = (
+  originalForm: IPopulatedForm,
+  editFormFieldParams: EditFormFieldParams,
+): ResultAsync<
+  IPopulatedForm,
+  EditFieldError | ReturnType<typeof transformMongoError>
+> => {
+  // TODO(#815): Split out this function into their own separate service functions depending on the update type.
+  return getUpdatedFormFields(
+    originalForm.form_fields,
+    editFormFieldParams,
+  ).asyncAndThen((newFormFields) => {
+    // Update form fields of original form.
+    originalForm.form_fields = newFormFields
+    return ResultAsync.fromPromise(originalForm.save(), (error) => {
+      logger.error({
+        message: 'Error encountered while editing form fields',
+        meta: {
+          action: 'editFormFields',
+          originalForm,
+          editFormFieldParams,
+        },
+        error,
+      })
+
+      return transformMongoError(error)
+    })
+  })
+}
+
+/**
+ * Updates form schema of given form depending on the given formUpdateParams
+ * @param originalForm the original form to update form fields for
+ * @param formUpdateParams the parameters to update the original form with
+ *
+ * @returns ok(updated form) if form fields update successfully
+ * @returns err(set of DatabaseError) if any database errors occurs
+ */
+export const updateForm = (
+  originalForm: IPopulatedForm,
+  formUpdateParams: Except<FormUpdateParams, 'editFormField'>,
+): ResultAsync<IPopulatedForm, ReturnType<typeof transformMongoError>> => {
+  // TODO(#815): Split out this function into their own separate service functions depending on the form parameter to update.
+  // Scrub formUpdateParams to remove private details, if any.
+  const scrubbedFormParams = omit(formUpdateParams, [
```

Comment:
instead of using `omit` and then `assignIn`, might it be better to be more explicit about which keys we're assigning? the `assignIn` seems to obscure the modifications being made to the form

Reply:
We don't know which keys are being assigned though, this depends on what the frontend sends back.

Reply:
okay, I suppose we can defer this to when we shard the update endpoint